File: 9-palindrome-number/cpp/main.cpp
class Solution {
 public:
  bool isPalindrome(int x) {
    if (x < 0) return false;
    return reverseInt(x) == x;
  }

  long reverseInt(long x) {
    long res = 0;
    while (x > 0) {
      res *= 10;
      res += x % 10;
      x /= 10;
    }

    return res;
  }
};

File: 9-palindrome-number/go/main.go
package main

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	return reverseInt(x) == x
}

func reverseInt(x int) int {
	rev := 0

	for x > 0 {
		rev = rev*10 + x%10
		x /= 10
	}
	return rev
}
